flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBMemoryControllerUtilsTest.java

```diff
@@ -91,10 +93,33 @@ public void testCreateSharedResourcesWithExpectedCapacity() {
 		long totalMemorySize = 2048L;
 		double writeBufferRatio = 0.5;
 		double highPriPoolRatio = 0.1;
-		RocksDBMemoryControllerUtils.allocateRocksDBSharedResources(totalMemorySize, writeBufferRatio, highPriPoolRatio);
+		RocksDBSharedResources rocksDBSharedResources = RocksDBMemoryControllerUtils.allocateRocksDBSharedResources(totalMemorySize, writeBufferRatio, highPriPoolRatio);
 		long expectedCacheCapacity = RocksDBMemoryControllerUtils.calculateActualCacheCapacity(totalMemorySize, writeBufferRatio);
 		long expectedWbmCapacity = RocksDBMemoryControllerUtils.calculateWriteBufferManagerCapacity(totalMemorySize, writeBufferRatio);
+
 		assertThat(actualCacheCapacity.get(), is(expectedCacheCapacity));
 		assertThat(actualWbmCapacity.get(), is(expectedWbmCapacity));
+		assertThat(rocksDBSharedResources.getWriteBufferManagerCapacity(), is(expectedWbmCapacity));
+	}
+
+	@Test
+	public void testCalculateRocksDBDefaultArenaBlockSize() {
+		long writeBufferSize = 64 * 1024 * 1024;
+		long expectArenaBlockSize = writeBufferSize / 8;
+		assertThat(RocksDBMemoryControllerUtils.calculateRocksDBDefaultArenaBlockSize(writeBufferSize), is(expectArenaBlockSize));
+	}
+
+	@Test
+	public void testCalculateRocksDBMutableLimit() {
+		long bufferSize = 64 * 1024 * 1024;
+		long limit = bufferSize * 7 / 8;
+		assertThat(RocksDBMemoryControllerUtils.calculateRocksDBMutableLimit(bufferSize), is(limit));
+	}
+
+	@Test
+	public void testValidateArenaBlockSize() {
+		long arenaBlockSize = 8 * 1024 * 1024;
+		assertFalse(RocksDBMemoryControllerUtils.validateArenaBlockSize(arenaBlockSize, (long) (arenaBlockSize * 0.5)));
+		assertTrue(RocksDBMemoryControllerUtils.validateArenaBlockSize(arenaBlockSize, (long) (arenaBlockSize * 1.5)));
 	}
```

Comment:
I'm hesitating on adding these tests since they're testing against the implementation instead of any contract. Once the calculation formula changes, these tests will fail and need to be adjusted accordingly.

Reply:
The contract for e.g. `calculateRocksDBDefaultArenaBlockSize` would be what the JavaDoc says? What a test for `calculateRocksDBDefaultArenaBlockSize` should do IMHO is to test that the method follows its documentation. If the formula changes, the JavaDoc should change and thus the test method would change too. Does this make sense?

BTW, now when I look at this closer, I think there's a minor issue. In https://github.com/ververica/frocksdb/blob/49bc897d5d768026f1eb816d960c1f2383396ef4/db/column_family.cc#L200 there's the aligning that is missing from the method. Also the test should check a case when alignment is needed and where it's not needed. I'll fix this later.

Reply:
I added testing the alignment to `testCalculateRocksDBDefaultArenaBlockSize`.

Reply:
What do you think now @carp84 ?

Reply:
ok, I'm fine to keep these tests. And +1 for adding the alignment logic, please check my inline comments. Thanks.